lib/org/chromium/webidl/HTMLFileContent.js

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'HTMLFileContent',
+
+  requires: [
+    'foam.parsers.HTMLLexer'
+  ],
+
+  properties: [
+    {
+      name: 'file',
+      // description: 'content of the HTML file',
+      postSet: function(_, file) {
+        this.pre = this.parse(file);
+      }
+    },
+    {
+      name: 'pre',
+      // description: 'all pre tags found in the parsed HTML file',
+    }
+  ],
+
+  methods: [
+    function parse(str) {
+      var self = this;
+      var lexer = self.HTMLLexer.create();
+      var OPEN = lexer.TagType.OPEN.name;
+      var CLOSE = lexer.TagType.CLOSE.name;
+      var OPEN_CLOSE = lexer.TagType.OPEN_CLOSE.name;
+
+      var results = lexer.parseString(str).value;
+
+      // Tag matching
+      // Using a stack to obtain ancestorOf <pre class=idl> tags
+      var tags = [];
+      var pre = [];
+      results.forEach(function(node, i) {
+        var item = results[i];
+
+        if (foam.core.FObject.isInstance(item)) {
+          var top = tags[tags.length - 1];
+          if (top === undefined || item.type.name === OPEN) {
+            tags.push(item);
+          } else if (item.nodeName === top.nodeName && item.type.name === CLOSE) {
+            // Do we need to do anything with this?
+            tags.pop();
+          } else if (item.type.name === OPEN_CLOSE && item.nodeName === 'pre') {
+            // Determine if node is of class IDL
+            var isIDL = false;
+            item.attributes.forEach(function(attr) {
+              if (attr.name === 'class' && attr.value.split(' ').includes('idl')) {
```

Comment:
I thought about this for a little and decided to implement it slightly differently. 

Scanning the stack every time we have a potential tag we are interested in seemed like it could be very costly (e.g. when there is deep nesting of tags occurring). So I am using a variable (`skipStack`) to keep track of the level in the stack where these excluded tags are included. It itself behaves as a stack as we may potentially have nesting of excluded tags.

Please let me know if you see any flaws / problems with this approach.